runtime/runtime_controller.h

```diff
@@ -41,53 +117,329 @@ class RuntimeController final : public WindowClient {
 
+  //----------------------------------------------------------------------------
+  /// @brief      Dart code cannot fully measure the time it takes for a
+  ///             specific frame to be rendered. This is because Dart code only
+  ///             runs on the UI task runner. That is only a small part of the
+  ///             overall frame workload. The GPU task runner frame workload is
+  ///             executed on a thread where Dart code cannot run (and hence
+  ///             instrument). Besides, due to the pipelined nature of rendering
+  ///             in Flutter, there may be multiple frame workloads being
+  ///             processed at any given time. However, for non-Timeline based
+  ///             profiling, it is useful for trace collection and processing to
+  ///             happen in Dart. To do this, the GPU task runner frame
+  ///             workloads need to be instrumented separately. After a set
+  ///             number of these profiles have been gathered, they need to be
+  ///             reported back to Dart code. The engine reports this extra
+  ///             instrumentation information back to Dart code running on the
+  ///             engine by invoking this method at predefined intervals.
+  ///
+  /// @see        `Engine::ReportTimings`, `FrameTiming`
+  ///
+  /// @param[in]  timings  Collection of `FrameTiming::kCount` * `n` timestamps
+  ///                      for `n` frames whose timings have not been reported
+  ///                      yet. A collection of integers is reported here for
+  ///                      easier conversions to Dart objects. The timestamps
+  ///                      are measured against the system monotonic clock
+  ///                      measured in microseconds.
+  ///
   bool ReportTimings(std::vector<int64_t> timings);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Notify the Dart VM that no frame workloads are expected on the
+  ///             UI task runner till the specified deadline. The VM uses this
+  ///             opportunity to perform garbage collection operations is a
+  ///             manner that interferes as little as possible with frame
+  ///             rendering.
+  ///
+  /// NotifyIdle is advisory and not making the same or making it with
+  /// insufficient deadlines does not mean that the VM will keep consuming
+  /// memory indefinitely. Even if the engine made absolutely no calls to
+  /// NotifyIdle, the VM would still perform garbage collection. The only issue
+  /// in such a world would be that the GC pause would happen in the middle of a
+  /// frame workload.
+  ///
+  /// The garbage collection mechanism and its thresholds are internal
+  /// implementation details and absolutely no guarantees are made about the
+  /// threshold discussed below. This discussion is also an oversimplification
+  /// but hopefully serves to calibrate expectations about GC behavior:
```

Comment:
Is there Dart documentation we could link to here instead of all of this?

People who read this will probably think it's more precise than it is, and it seems likely that this could drift out of sync with actual implementation details it's trying to capture.

Reply:
Alternatively, would it make sense to move some of this to somewhere in //dart/runtime/vm where it could be maintained by the VM team? It is a really helpful description.

Reply:
> Is there Dart documentation we could link to here instead of all of this?

No. It's not documented anywhere. And probably for the same reasons we hesitate to document it here. It's implementation details and may change.

> People who read this will probably think it's more precise than it is, and it seems likely that this could drift out of sync with actual implementation details it's trying to capture.

Fair point. But the caveats (stuff like "are implementation details and absolutely no guarantees ... oversimplification" ) I believe are sufficient for reasonable readers to not make assumptions of stability.